packages/connect-popup/src/index.tsx

```diff
@@ -360,6 +360,18 @@ const handleLogMessage = (event: MessageEvent<IFrameLogRequest>) => {
 
 const handleContentScriptLoaded = (data: PopupContentScriptLoaded) => {
     log.debug('content-script loaded', data.payload);
+    // Check if extension ID matches the popup URL
+    const urlParams = new URLSearchParams(window.location.search);
+    const targetExtensionId = urlParams.get('extension-id');
+    console.log('popup targetExtensionId', targetExtensionId, 'data.payload.id', data.payload.id);
+    if (targetExtensionId && targetExtensionId !== data.payload.id) {
+        return;
+    }
+    if (data.payload.contentScriptVersion !== CONTENT_SCRIPT_VERSION) {
+        console.error(
+            'Content script version mismatch! Make sure to update content script to the latest version from @trezor/connect-webextension.',
+        );
```

Comment:
couldn't this warning be added also to the client side? I think it would be more visible there for 3rd party devs. something like this line maybe? https://github.com/trezor/trezor-suite/blob/develop/packages/connect-webextension/src/contentScript.ts#L36

Reply:
Good catch, actually it's a bad idea to put it here since the production version can be different.